File: src/main/java/com/xellitix/commons/semver/metadata/IdentifierFactory.java
package com.xellitix.commons.semver.metadata;

/**
 * {@link Identifier} factory.
 *
 * @author Grayson Kuhns
 */
public interface IdentifierFactory {

  /**
   * Creates an {@link Identifier}.
   *
   * @param value The value.
   * @return The {@link Identifier}.
   */
  Identifier create(String value);

  /**
   * Creates an {@link Identifier}.
   *
   * @param value The value.
   * @return The {@link Identifier}.
   */
  Identifier create(Integer value);

  /**
   * Creates an {@link Identifier}.
   *
   * @param value The value.
   * @return The {@link Identifier}.
   */
  Identifier create(Long value);
}

File: src/main/kotlin/com/xellitix/commons/semver/metadata/IdentifierFactory.kt
package com.xellitix.commons.semver.metadata

import com.google.inject.Singleton

/**
 * [Identifier] factory.
 */
@Singleton
class IdentifierFactory {

    /**
     * Creates an [Identifier].
     */
    fun create(value: String): Identifier {
        return Identifier(value)
    }

    /**
     * Creates an [Identifier].
     */
    fun create(value: Int): Identifier {
        return Identifier(value)
    }

    /**
     * Creates an [Identifier].
     */
    fun create(value: Long): Identifier {
        return Identifier(value)
    }
}
